examples/syntest.rs

```
@@ -0,0 +1,247 @@
+                            scopes_on_line_being_tested.push(
+                                ScopedText {
+                                    char_start: col,
+                                    text_len: len,
+                                    scope: stack.as_slice().to_vec()
+                                }
+                            );
+                            col += len;
+                        }
+                    }
+                }
+                
+                line.clear();
+                current_line_number += 1;
+                if reader.read_line(&mut line).unwrap() == 0 {
+                    break;
+                }
+                line = line.replace("\r", &"");
+            }
+            Ok(
+                if assertion_failures > 0 {
+                    SyntaxTestFileResult::FailedAssertions(assertion_failures, total_assertions)
+                } else {
+                    SyntaxTestFileResult::Success(total_assertions)
+                }
+            )
+        },
+        None => Err(SyntaxTestHeaderError::MalformedHeader)
+    }
+}
+
+fn main() {
+    let ss = SyntaxSet::load_defaults_newlines(); // note we load the version with newlines
+
+    let args: Vec<String> = std::env::args().collect();
+    let path = if args.len() < 2 {
+        "."
+    } else {
+        &args[1]
+    };
+
+    let exit_code = recursive_walk(&ss, &path);
+    std::process::exit(exit_code);
+
+}
+
+
+fn recursive_walk(ss: &SyntaxSet, path: &str) -> i32 {
+    let mut exit_code: i32 = 0;
+    let walker = WalkDir::new(path).into_iter();
+    for entry in walker.filter_entry(|e|e.file_type().is_dir() || is_a_syntax_test_file(e)) {
+        let entry = entry.unwrap();
+        if entry.file_type().is_file() {
+            println!("Testing file {}", entry.path().display());
+            let result = test_file(&ss, entry.path(), true);
+            println!("{:?}", result);
+            if let Err(_) = result {
```

it currently doesn't die on the first failing file, it just ensures that when it has run through all the files, it will exit with an appropriate exit code :) maybe we could add a flag though so that it optionally can fail fast, to save time (not that these syntax tests take long at all!) when all the caller wants to know is whether there were any failures